### ISSUE TITLE
kivy-ios/dist/hostpython2/bin/python: No such file or directory

### ISSUE THREAD

**Issue by irreverentperson** [kivy/kivy-ios#319]
MacOS 10.13.6
Xcode 10.1

Just recloned kivy-ios and did clean install of everything, linked libraries, disabled bitcode.
When i hit play in xcode i get the following error.

/Users/jim/Library/Developer/Xcode/DerivedData/tester-epnmulbhkafyfbdrxbfkcscnxfwh/Build/Intermediates.noindex/tester.build/Debug-iphoneos/tester.build/Script-113D17E2153E3DB5001310A6.sh: line 2: /Users/jim/kivy-ios/dist/hostpython2/bin/python: No such file or directory

I specified the python version
./toolchain.py build hostpython2
./toolchain.py build python2
./toolchain.py build kivy

then create the project and getting the same error on play.
the folder is named kivy-ios/dist/hostpython. not hostpython2.
which is why it's saying there's no such directory.

**Comment by HeaTTheatR** [comment]
@irreverentperson - https://github.com/kivy/kivy-ios/issues/312

**Comment by hackalog** [comment]
Yes, I see the same error on one of my builds (directory is called hostpython and not hostpython2)

@irreverentperson any reason you closed your other thread?

**Comment by hoangvuche** [comment]
the same problem when i tried with MacOS 10.14.2 and Xcode 10.1

**Comment by hackalog** [comment]
@hoangvuche, @HeaTTheatR, @irreverentperson, can you please try PR#331 and see if it fixes this issue for you?

**Comment by hackalog** [comment]
@hoangvuche Did you clean out the built recipes before you tried that?
```
# (ignored) Download python2
# (ignored) Extract python2
# (ignored) Build_all python2
```
The run you attached seems to say you had a cached build

Can you try again after cleaning your kivy-ios build first? i.e.

`./toolchain.py status`

and then

`.toolchain.py clean hostpython2 python2`  # and any additional package that showed as built, after the `status` command above

Thanks!

**Comment by tito** [comment]
The Xcode template also need to be updated, as both python2 and python3 install in the same "hostpython" directory. This is why you have the issue

**Comment by tito** [comment]
Fixed via #331

**Comment by hoangvuche** [comment]
Many thanks @hackalog
I built successfully after cleaning built packages as your guidance.
